stablehlo/reference/Index.h

```diff
@@ -22,10 +22,17 @@ limitations under the License.
 #include "llvm/ADT/Optional.h"
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/Support/Error.h"
+#include "mlir/Support/LogicalResult.h"
 
 namespace mlir {
 namespace stablehlo {
 
+/// Check if the 'index' is a valid index in the index space of a tensor with
+/// shape 'shape'. Specifically, for a shape '(d0)x(d1)x...x(dR-1)' and an index
+/// '{i0, i1, ..., iR-1}', we check if 0 <= i[k] <= d[k] for k in
+/// {0, 1, ..., R-1}. Note that the check also implies that 'd[k]' >= 1.
```

Comment:
That's a good point! Sounds like `index_begin` will now start crashing for tensors with dimensions equal to zero? Let's update `index_begin` to return `{}` in that case and update tests accordingly?